File: SIN BDD/SOAP_DOTNET_CONUNI_GR06/07. CLIMOVIL/cliente_movil_soap_dotnet_conuni_gr06/src/Servicio/ServicioAutenticacion.cs
namespace Ec.Edu.Monster.Servicio;

public class ServicioAutenticacion
{
    private readonly Utilidades.ClienteSoap cliente = new();

    public bool Autenticar(string usuario, string contrasena)
    {
        return cliente.Ejecutar(servicio => servicio.IniciarSesion(usuario, contrasena));
    }
}

File: SIN BDD/RESTFULL_JAVA_CONUNI_GR06/07. CLIMOVIL/cliente_movil_restful_java_conuni_gr06/app/src/main/java/ec/edu/monster/servicio/ServicioAutenticacion.java
package ec.edu.monster.servicio;

import ec.edu.monster.util.ClienteRest;
import org.json.JSONObject;

public class ServicioAutenticacion {

    public boolean iniciarSesion(String usuario, String contrasena) throws Exception {
        JSONObject cuerpo = new JSONObject();
        cuerpo.put("usuario", usuario == null ? "" : usuario);
        cuerpo.put("contrasena", contrasena == null ? "" : contrasena);
        JSONObject respuesta = ClienteRest.post("/sesion", cuerpo.toString());
        return respuesta.optBoolean("valido", false);
    }
}
